### PR TITLE
Fire PlatformViewController FlutterView callbacks

### PR DESCRIPTION
**Pull request by mklim**
Fixes a bug where `PlatformViewController` was not being notified of `FlutterView` attachment changes.

Related to flutter/flutter#41851.

### CODE REVIEW

**Comment by mklim** [comment]
@amirh this still needs a test before it can land but if you want to review the basic approach in the meantime feel free.

**Comment by mklim** [comment]
This is ready for review. The unit test I added is not good, it's just an interaction test. To really test the behavior I would have need to have significantly refactored the `PlatformViewsController` class and then changed `FlutterEngine` to use constructor injection. That may be worth it, but I wasn't sure whether it was good to make this PR any riskier.

File: shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java [hunk 1]
```diff
@@ -121,7 +121,7 @@ public static native void nativeInit(
 
   // TODO(mattcarroll): add javadocs
   @UiThread
-  public static native boolean nativeGetIsSoftwareRenderingEnabled();
+  public native boolean nativeGetIsSoftwareRenderingEnabled();
```
Comment (dnfield) [inline shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java]: Was this meant to be part of this change?

Comment (dnfield) [reply shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java]: I see now.

This seems to indicate that most of these methods shouldn't be static then, right?

File: shell/platform/android/test/io/flutter/embedding/android/FlutterViewTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     FlutterView flutterView = new FlutterView(RuntimeEnvironment.application);
+     FlutterEngine flutterEngine = spy(new FlutterEngine(RuntimeEnvironment.application, mockFlutterJni));
+     when(flutterEngine.getPlatformViewsController()).thenReturn(platformViewsController);
+
+     flutterView.attachToFlutterEngine(flutterEngine);
+     flutterView.detachFromFlutterEngine();
+
+     verify(platformViewsController, times(1)).detachFromView();
+   }
+}
```
Comment (dnfield) [inline shell/platform/android/test/io/flutter/embedding/android/FlutterViewTest.java]: nit: new line at EOF

**Comment by mklim** [comment]
> I suspect that many of the methods on FlutterJNI should be made instance methods similar to this one, except for ones that we know will always be global (like the observatory uri).

I agree. I conservatively just instanced the one thing I needed to for this test case but I think it would make sense to move all of them. These are really depending on some mutable state (the native library being loaded) so I don't think they're really immutable/"static" to begin with.

**Pull request merged**